modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -182,6 +238,175 @@ MechanicalContactConstraint::jacobianSetup()
+            else
+              pinfo->_mech_status = PenetrationInfo::MS_SLIPPING_FRICTION;
+          }
+          else
+          {
+            pinfo->_mech_status = PenetrationInfo::MS_STICKING;
+            pinfo->_lagrange_multiplier_slip += penalty_slip * tangential_inc_slip;
+          }
+        }
+      }
+    }
+  }
+}
+
+bool
+MechanicalContactConstraint::AugmentedLagrangianContactConverged()
+{
+  Real contactResidual = 0.0;
+
+  Real converged = 0.0;
+
+  for (auto & pinfo_pair : _penetration_locator._penetration_info)
+  {
+    const dof_id_type slave_node_num = pinfo_pair.first;
+    PenetrationInfo * pinfo = pinfo_pair.second;
+
+    // Skip this pinfo if there are no DOFs on this node.
+    if (!pinfo || pinfo->_node->n_comp(_sys.number(), _vars[_component]) < 1)
+      continue;
+
+    const Real distance = pinfo->_normal * (pinfo->_closest_point - _mesh.nodeRef(slave_node_num));
+
+    if (pinfo->isCaptured())
+    {
+
+      if (contactResidual < std::abs(distance))
+        contactResidual = std::abs(distance);
+
+      // penetration < tol
+      if (contactResidual > _al_penetration_tolerance)
+      {
+        _console << "Augmented Lagrangian contact penetration enforcement is NOT satisfied \n";
+        converged = 1;
+        break;
+      }
+
+      if (_model == CM_COULOMB)
+      {
+        RealVectorValue contact_force_normal((pinfo->_contact_force * pinfo->_normal) *
+                                             pinfo->_normal);
+        RealVectorValue contact_force_tangential(pinfo->_contact_force - contact_force_normal);
+
+        RealVectorValue tangential_inc_slip =
+            pinfo->_incremental_slip - (pinfo->_incremental_slip * pinfo->_normal) * pinfo->_normal;
+
+        const Real tan_mag(contact_force_tangential.norm());
+        const Real tangential_inc_slip_mag = tangential_inc_slip.norm();
+
+        RealVectorValue distance_vec =
+            (pinfo->_normal * (_mesh.nodeRef(slave_node_num) - pinfo->_closest_point)) *
+            pinfo->_normal;
+
+        Real penalty = getPenalty(*pinfo);
+        RealVectorValue pen_force_normal =
+            penalty * distance_vec + pinfo->_lagrange_multiplier * pinfo->_normal;
+
+        // Frictional capacity
+        Real capacity(_friction_coefficient * (pen_force_normal * pinfo->_normal < 0
+                                                   ? -pen_force_normal * pinfo->_normal
+                                                   : 0.0));
+
+        // incremental slip <= tol for all pinfo_pair such that tan_mag < capacity
+        if (MooseUtils::absoluteFuzzyLessThan(tan_mag, capacity) &&
+            pinfo->_mech_status == PenetrationInfo::MS_STICKING)
+        {
+          if (MooseUtils::absoluteFuzzyGreaterThan(tangential_inc_slip_mag,
+                                                   _al_incremental_slip_tolerance))
+          {
+            _console << "Augmented Lagrangian contact tangential sliding enforcement is NOT "
```

Comment:
This stuff won't go to the console if you hit it on a processor other than processor 0.